### PR TITLE
[Core] Add missing spec for ContainsProductRuleChecker

### PR DESCRIPTION
**Pull request by PWalkow**
| Q             | A
| ------------- | ---
| Bug fix?      | [no]
| New feature?  | [no]
| BC breaks?    | [no]
| Deprecations? | [no]
| Fixed tickets | []
| License       | MIT
| Doc PR        | [missing Specs for RuleChecker]

### CODE REVIEW

File: src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Checker\ContainsProductRuleChecker');
+    }
+
+    function it_should_be_Sylius_rule_checker()
+    {
+        $this->shouldImplement('Sylius\Component\Promotion\Checker\RuleCheckerInterface');
+    }
+
+    function it_throw_exception_on_invalid_subject(PromotionSubjectInterface $subject)
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php]: ``throws``

File: src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * @author Piotr Walków <walkow.piotr@gmail.com>
+ */
+class ContainsProductRuleCheckerSpec extends ObjectBehavior
+{
+    function it_should_be_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Checker\ContainsProductRuleChecker');
+    }
+
+    function it_should_be_Sylius_rule_checker()
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php]: ``it_is``

File: src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+use Sylius\Component\Core\Model\ProductVariant;
+use Sylius\Component\Promotion\Exception\UnsupportedTypeException;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+/**
+ * @author Piotr Walków <walkow.piotr@gmail.com>
+ */
+class ContainsProductRuleCheckerSpec extends ObjectBehavior
+{
+    function it_should_be_initializable()
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php]: ``it_is``

File: src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        $this->shouldThrow(UnsupportedTypeException::class)->duringIsEligible($subject, []);
+    }
+
+    function it_returns_true_if_variant_is_right_and_exclude_is_not_set(
+        OrderInterface $subject,
+        OrderItem $orderItem,
+        ProductVariant $variant
+    ) {
+        $subject->getItems()->shouldBeCalled()->willReturn([$orderItem]);
```
Comment (stloyd) [inline src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php]: `shouldBeCalled()` is useless when you use `willReturn()`. Don't use mocks & stubs to test same thing!

Comment (pamil) [inline src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php]: `shouldBeCalled()` says explicitly that some method should be called (spec should be as readable as possible) and gives us faster and direct feedback if something that should be called hasn't been called.

Comment (stloyd) [inline src/Sylius/Component/Core/spec/Promotion/Checker/ContainsProductRuleCheckerSpec.php]: Then you misunderstand purposes of those two things, I have already explained it some time ago but don't have time now to do this again (as well for looking across the PRs to find those notes), please read:
- http://adamwathan.me/2014/09/22/when-to-mock/
- some parts of this http://everzet.com/post/33178339051/sus-collaborators cover that topic too

**Comment by PWalkow** [comment]
Removed 'shouldBeCalled'

**Comment by pjedrzejewski** [comment]
Thanks @PWalkow!

**Pull request merged**